packages/core/src/ui/components/DappTransaction/DappTransaction.tsx

```diff
@@ -1,91 +1,84 @@
+/* eslint-disable sonarjs/no-duplicate-string */
 import React from 'react';
-import { Ellipsis, ErrorPane } from '@lace/common';
+import { ErrorPane } from '@lace/common';
+import { Wallet } from '@lace/cardano';
 import { DappInfo, DappInfoProps } from '../DappInfo';
+import { DappTxHeader } from './DappTxHeader/DappTxHeader';
+import { DappTxAsset } from './DappTxAsset/DappTxAsset';
+import { DappTxOutput } from './DappTxOutput/DappTxOutput';
 import styles from './DappTransaction.module.scss';
-import { TranslationsFor } from '@ui/utils/types';
-
-type TransactionDetails = {
-  fee: string;
-  outputs: {
-    coins: string;
-    recipient: string;
-    assets?: {
-      name: string;
-      amount: string;
-      ticker?: string;
-    }[];
-  }[];
-  type: 'Send' | 'Mint' | 'Burn';
-};
+import { useTranslate } from '@src/ui/hooks';
+import { TransactionFee } from '@ui/components/ActivityDetail';
 
 export interface DappTransactionProps {
   /** Transaction details such as type, amount, fee and address */
-  transaction: TransactionDetails;
+  transaction: Wallet.Cip30SignTxSummary;
   /** dApp information such as logo, name and url */
   dappInfo: Omit<DappInfoProps, 'className'>;
   /** Optional error message */
   errorMessage?: string;
-  translations: TranslationsFor<'transaction' | 'amount' | 'recipient' | 'fee' | 'adaFollowingNumericValue'>;
+  fiatCurrencyCode?: string;
+  fiatCurrencyPrice?: number;
+  coinSymbol?: string;
 }
 
 export const DappTransaction = ({
-  transaction: { type, outputs, fee },
+  transaction: { type, outputs, fee, mintedAssets, burnedAssets },
   dappInfo,
   errorMessage,
-  translations
-}: DappTransactionProps): React.ReactElement => (
-  <div>
-    <DappInfo {...dappInfo} className={styles.dappInfo} />
-    {errorMessage && <ErrorPane error={errorMessage} className={styles.error} />}
-    <div data-testid="dapp-transaction-container" className={styles.details}>
-      <div className={styles.header}>
-        <div data-testid="dapp-transaction-title" className={styles.title}>
-          {translations.transaction}
-        </div>
-        <div data-testid="dapp-transaction-type" className={styles.type}>
-          {type}
-        </div>
+  fiatCurrencyCode,
+  fiatCurrencyPrice,
+  coinSymbol
+}: DappTransactionProps): React.ReactElement => {
+  const { t } = useTranslate();
+  return (
+    <div>
+      <DappInfo {...dappInfo} className={styles.dappInfo} />
+      {errorMessage && <ErrorPane error={errorMessage} className={styles.error} />}
+      <div data-testid="dapp-transaction-container" className={styles.details}>
+        {type === 'Mint' && mintedAssets?.length > 0 && (
```

Comment:
could we do something like:
```suggestion
        {type === Wallet.Cip30TxType.Mint && mintedAssets?.length > 0 && (
```

Reply:
Good catch. I overlooked it during migration. We have so much duplication of types across extension and core/cardano packages. Let's improve it.